### ISSUE TITLE
Support not overwritting existing rows

### ISSUE THREAD

**Issue by gregsterin** [hortonworks-spark/shc#144]
Currently, it looks like the SaveMode passed from spark is ignored. Theres an argument in insert:

https://github.com/hortonworks-spark/shc/blob/2b5eba6cebe832a1f5b0b51d41b2ee64415cfb2e/core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseRelation.scala#L177

But it doesn't seem to be used.
Overwrite happens regardless of SaveMode.

Is there an opportunity to respect the SaveMode here?

**Comment by weiqingy** [comment]
Right. SHC may fully support SaveMode in the near future. Welcome any PRs for this.

**Comment by adamdec** [comment]
Actually now it appends...not overwrites.
